app/state.go

```diff
@@ -1835,3 +1843,207 @@ func setBrokenArtifactFlag(store store.Store, artName string) {
 		// already in an error path.
 	}
 }
+
+type controlMapState struct {
+	wrappedState UpdateState
+	baseState
+}
+
+func NewControlMapState(wrapsState UpdateState) State {
```

Comment:
From looking at your new state machine failsafe PR, these should now implement `UpdateState` right?